NebulaWorld/Player/DroneManager.cs

```diff
@@ -9,16 +9,26 @@ public static class DroneManager
         public static int[] DronePriorities = new int[255];
         public static Random rnd = new Random();
         public static Dictionary<ushort, List<int>> PlayerDroneBuildingPlans = new Dictionary<ushort, List<int>>();
+        public static List<int> PendingBuildRequests = new List<int>();
```

Comment:
Shouldn't this be locked?

Reply:
I believe it is modified only from the main thread, from the `GameTick()` or from the `Update()` when packets are processed. It is purely used only on the client-side.

Reply:
I might be thinking about it too much, but there's not much clarity or convention on when something can be accessed from Host/Client/Both and what is the source of truth for each one.

Reply:
For this case of `PendingBuildRequests` it is used by clients to remember, for which prebuilds the request was sent. The issue was following: drone reached the prebuild, client sent request to build it, however from the drone point of view the prebuild still exist and still need to be built. So I have created this list to ignore prebuilds for which the client already sent request and it is just waiting for the response from server. 

But it is true, that for the "Manager" classes, it is sometimes hard to see, what is used by clients and what by the host.